crates/dc_jni/src/jni.rs

```diff
@@ -234,6 +234,11 @@ fn jni_add_nodes<'local>(
                             );
                         }
                     }
+                    for LayoutParentChildren { parent_layout_id, child_layout_ids } in
+                        &node_list.parent_children
+                    {
+                        manager.update_children(*parent_layout_id, child_layout_ids)
```

Comment:
Can you remind me why this change is needed? Was this to avoid a problem when views were added out of order?

Reply:
Yeah, it does a pass to "update the native layout tree" once it's built its full tree in Kotlin, and I couldn't think of an efficient way to visit the subset of the tree that had changed (so I could use the old index-based child update methods) in the right order. So instead, I made it so that the parent would just specify all of it's child indices whenever there's a change, and then the algorithm to figure out how to update only the invalidated layout nodes became simple (it just visits the invalidated nodes in tree order).

This way the caching that the Rust layout engine does gets used, so the rust layout iterations are faster.